src/modules/tensor/hip/kernel/tensor_sum.cpp

```
@@ -1212,7 +1212,7 @@ RppStatus hip_exec_tensor_sum(T *srcPtr,
         Rpp32u partialSumArrLength = gridDim_x * gridDim_y * gridDim_z;
         float *partialSumArr;
         partialSumArr = handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem;
```

partialSumArr gets allocated in scratch_mem without even checking if there is enough memory available. This is very dangerous and can cause mem allocation failures. We need to get rid of this kind of memory allocation. We would like to catch failure on allocation and gracefully exit if it happens instead of not assuming any bounds for memory